### ISSUE TITLE
QT checks are too stringent

### ISSUE THREAD

**Issue by ppavlidis** [PavlidisLab/Gemma#617]
Currently regenerating diagnostics from the GUI will fail if the QT fails checks. Having to bypass this on the CLI is annoying.

This behaviour should be reconsidered. It's fine to generate a warning.

**Comment by ppavlidis** [comment]
/When examining distributions of values, I would be careful using the mean, max or min as descriptive statistics. If the intention is to find symmetry around 0, use the median. For examining extremes, use the +/-95%ile (or maybe 90%).

**Comment by arteymix** [comment]
I agree with the first point, these checks were not intended for the GUI, or at least not if there is no option to override from there.

As to the second point, relaxing the checks by using robust statistics and maybe laxer thresholds seem to be a good idea. Note that I based those on thousands of datasets we already have curated correctly.

**Comment by ppavlidis** [comment]
Yeah I was basing this on the data in our cleanup spreadsheet. You can look at that to see the (apparent) false positives.

**Comment by ppavlidis** [comment]
Modifications to be done:
* "Linear with negatives" this is okay because background subtraction can yield negative values. As long as the median value is not very near zero this is not an issue. If the distribution is symmetric(ish) around zero, it might actually be ratios. 
* "Too large" for log2-scaled data when value is over ~16. This is not a problem. Values over 20 would start to raise more suspicion.  The reason for variance from the usually-expected 2^16 limit is there is a lot of variety in how people normalize their data.

(These limits should never be examined for data we have reprocessed from raw i.e. it is not 'external', not sure if that is the case)

**Comment by arteymix** [comment]
There's a check that exclude background subtracted and normalized datasets from the checks. Is the raw data you tested properly annotated?

**Comment by ppavlidis** [comment]
This brings up a good point.

The background subtraction and normalization flags are inferred from the GEO annotation. Unfortunately the absence of those flags does not mean anything because GEO has no standards in how the data are described.

When we record True for these elements, it (usually) means the submitter explicitly says.

Typically but not always background subtraction and normalization are in fact true but we can't tell from the description.

This means the value of those flags has turned out to be low. They should not be relied on (for raw data especially)

**Comment by ppavlidis** [comment]
This in the logs:

```
ExpressionDataDoubleMatrixUtil.infer(348) | Data is normalized sample-wise: one or more sample has either zero mean or median.
```

This is misleading. It is SOP to normalize data sample-wise. The logging here is in reference to a particular type of normalization that centers the data on zero. The message (if it's even needed) should be clearer that it is referring to centering of data, not just sample-wise normalization.

**Comment by ppavlidis** [comment]
This is an RNA-seq study we analyzed ourselves, so it should pass the checks with flying colours. It does not, with it being detected as LOGBASEUNKNOWN, this should not happen and the -ignoreqm flag should not be needed for makeProcessedData.

https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?shortName=GSE157902.2

```
Caused by: ubic.gemma.core.datastructure.matrix.InferredQuantitationMismatchException: The scale of QuantitationType(super=QuantitationTypeImpl Id=562418 Name=log2cpm, isBackground=false, isBackgroundSubtracted=false, isBatchCorrected=false, isMaskedPreferred=tru\
e, isNormalized=false, isPreferred=true, isRecomputedFromRawData=true, isRatio=false, generalType=QUANTITATIVE, representation=DOUBLE, scale=LOG2, type=AMOUNT) differs from the one inferred from data: LOGBASEUNKNOWN.
        at ubic.gemma.core.datastructure.matrix.ExpressionDataDoubleMatrixUtil.ensureLog2Scale(ExpressionDataDoubleMatrixUtil.java:155) ~[gemma-core-1.29.9.jar:?]
```

**Comment by ppavlidis** [comment]
logging "detectSuspiciousValues(525) | Expression data is either background subtracted or normalized, suspicious values will not be linted." is too confusing (and may not be relevant). What is "suspicious" and what is "linted" in this context?

**Comment by ppavlidis** [comment]
Summarizing:

- [ ] Checks should be disabled for data that we reanalyzed (that is not 'external'); definitely `ignoreqm` should never be required. 
- [ ] Overall less logging associated with QT checking. It should be silent in normal conditions.
- [ ] Consider whether `ignoreqm` should be the default, because so far it is only generating false positives.

The reasoning for the last point is that in the future it will be increasingly rare that we are dealing with funky microarray studies, and the ones we are still dealing with would have already been subjected to QC checks. I'd rather have validation of the QTs be done by curators as part of their workflow and trust that when they run `makeProcessedData` they know what they are doing.

**Comment by ppavlidis** [comment]
Gemma 2 disposition: Won't fix
